core/src/Streamly/Internal/FileSystem/PosixPath.hs

```diff
@@ -366,17 +424,32 @@ toChars_ p =
 toString :: IsPath OS_PATH a => a -> [Char]
 toString = runIdentity . Stream.toList . toChars
 
--- | Decode the path to a Unicode string using strict CODEC_NAME decoding.
+-- | Decode the path to a Unicode string using lax CODEC_NAME decoding.
```

Comment:
Was the documentation initially incorrect? Or does this PR make it lax?

Reply:
You can see that this uses a lax unicode API, the doc was wrong.